javalib/src/main/scala/java/util/AbstractMap.scala

```diff
@@ -0,0 +1,199 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+object AbstractMap {
+
+  private def entryEquals[K, V](entry: Map.Entry[K, V], other: Any): Boolean = {
+    other match {
+      case other: Map.Entry[_, _] =>
+        entry.getKey == other.getKey && entry.getValue == other.getValue
```

Comment:
Ah, right. Prefer cast to `AnyRef`. But then better to do it only once in `sameValue`, and let it take `Any`s as parameters.